### ISSUE TITLE
모델) 회의록 코퍼스로 pre-training

### ISSUE THREAD

**Issue by HanNayeoniee** [boostcampaitech3/final-project-level3-nlp-09#39]
general domain 코퍼스로 학습된 RoBERTa-large 모델은 이번 프로젝트에서 학습에 사용하는 국회 회의록과 다른 특성(domain)을 가진다고 생각해 unlabeled 회의록 코퍼스로 추가 학습하고자 한다. 
특정 domain, task의 데이터로 추가 학습하는 방법의 효과에 대한 논문(**Don't Stop Pretraining**)은 [여기](https://www.notion.so/Don-t-Stop-Pretraining-Adapt-Language-Models-to-Domains-and-Tasks-a61a5c6aeac0456da624dbee0bd0e865)에 정리되어 있습니다.

**Comment by maxha97** [comment]
'Don't stop pretraining' 논문을 기반으로, 크게 도메인 기반 추가 Pretrained 학습인 DAPT(Domain-Adaptive Pretraining)와 태스크 기반 추가 Pretrained 학습인 TAPT(Task-Adaptive Pretraining)을 진행해볼 수 있었습니다. 
![image](https://user-images.githubusercontent.com/53209003/173003073-e486bc83-2573-4fca-8fae-69f255266448.png)

기존 부스트캠프 ODQA 대회에서, KorQuad 데이터로 진행한 TAPT의 경우 성능에 큰 향상이 없거나 떨어지는 효과를 보여 진행하지 않았습니다. 대신에 이번에 사용할 데이터 도메인이 기존에 QA 데이터셋에서 많이 활용되는 위키백과와는 형태가 많이 다르며, 이미 라벨링 되지 않은 국회 회의록 자료들이 많다는 사실에 기초하여 추가적인 Pretrained 학습을 진행하였습니다. 

방법은, Transformer 라이브러리에 제공하는 run_mlm.py 코드를 기준으로 현재 SOTA 모델이었던 Klue/Roberta-large 모델을 기준으로 Masked Language Modeling(MLM)을 진행하였습니다. Pretraining 특성상 시간이 오래 걸리기 때문에, 최대 에폭3까지 학습을 진행하였습니다. 또한, 회의록 본문 길이가 토큰 512개를 넘을 수 있음을 고려하여 문장 단위로 분리하여 학습을 진행하였습니다.

**Comment by maxha97** [comment]
실험 결과 
![image](https://user-images.githubusercontent.com/53209003/173006258-b2e84008-bb0b-4ea5-810d-12261536cfff.png)

**Comment by maxha97** [comment]
- Validation 성능 지표 그래프
![image](https://user-images.githubusercontent.com/53209003/173006258-b2e84008-bb0b-4ea5-810d-12261536cfff.png)

**Comment by maxha97** [comment]
### 실험 한계
실질적으로 실험을 진행하는데 한계가 있었으며, 이로 인한 영향일 수도 있을 것으로 예상됨
- 1에폭당 학습하는 시간이 약 6시간 정도 소요되었기 때문에, 논문과 같이 많은 에폭을 학습하기 어려웠음
- 데이터가 수치적으로 논문에 준하는 수준의 양을 갖추지 못해, 성능 향상에 한계가 있음
- QA 데이터로 Pretrain을 진행하여 성능이 오른 사례를 찾지 못하였기 때문에, 확실한 입증이 부족
